src/Generator/Generators/CSharp/CSharpMarshal.cs

```diff
@@ -580,13 +580,16 @@ public override bool VisitPointerType(PointerType pointer, TypeQualifiers quals)
                 if (isRefParam)
                 {
                     var typeName = Type.TypePrinterDelegate(finalPointee);
-
-                    if (param.IsInOut)
-                        Context.Before.WriteLine("{0} _{1} = {1};", typeName, param.Name);
+                    if (Context.Function.OperatorKind != CXXOperatorKind.Subscript)
+                    {
+                        if (param.IsInOut)
+                            Context.Before.WriteLine("{0} _{1} = {1};", typeName, param.Name);
```

Comment:
Now that you're changing this anyway, convert to interpolated strings this line and the other 3 in the changed block.